Automattic/wp-calypso
# Add Tracker guards to prevent using them when uninitialized

#### Proposed Changes

* Add guarding functions to trackers we have defined in the code
* Check if tracker is initialized as part of `mayWeTrackByTracker()`
* Create a separate function to check if we can initialize Tracker 
* Revert optional chaining of gtag usage

#### Testing Instructions

* Ensure `ad-tracking` is enabled on your development environment (PCYsg-dGP-p2#debugging)
* Comment out lines `client/lib/analytics/ad-tracking/setup.js:194-198`
* Launch Calypso checkout (http://calypso.localhost:3000/checkout/jetpack/jetpack_boost_yearly?flags=ad-tracking)
* Go through the purchase of the product, and monitor there are no error logs in console `window.gtag is not a function.`

#### Pre-merge Checklist





Related to #66493
